### PR TITLE
Validation updates to better support UTMs and regex rules

### PR DESCRIPTION
**Pull request by NickPhura**
# Overview

https://quartech.atlassian.net/browse/BHBC-1428
https://quartech.atlassian.net/browse/BHBC-1460

### Validation/Transformation
- Update format validation schema to support regex flags
  - Update validation schemas to have updated rules for waypoint and utm zone
  - Update spatial-utils to allow UTM zone letters to be optional
- Update transformation/validation schemas for both moose templates
  - Recruitment should now work 
- Update XLSX transformation logic to support better support conditional rules
  - This should also fix BHBC-1460

### N8N
- Pin n8n version to 0.144.0
- Update n8n workflow exports (latest n8n version has modified the config)

### Scraping Occurrences
- Update occurrence table to add `sex` column
- Update occurrence popup to include the sex field
- Tweak occurrence popup to better handle occurrences with no geometry

## This PR contains the following types of changes

- [ ] New feature (change which adds functionality)
- [x] Enhancement (improvements to existing functionality)
- [ ] Bug fix (change which fixes an issue)
- [ ] Misc cleanup / Refactoring / Documentation
- [ ] Version change

## Checklist

A list of items that are good to consider when making any changes.

_Note: this list is not exhaustive, and not all items are always applicable._

### General

- [ ] I have performed a self-review of my own code

### Code

- [ ] New files/classes/functions have appropriately descriptive names and comment blocks to describe their use/behaviour
- [ ] I have avoided duplicating code when possible, moving re-usable pieces into functions
- [ ] I have avoided hard-coding values where possible and moved any re-usable constants to a constants file
- [ ] My code is as flat as possible (avoids deeply nested if/else blocks, promise chains, etc)
- [ ] My code changes account for null/undefined values and handle errors appropriately
- [ ] My code uses types/interfaces to help describe values/parameters/etc, help ensure type safety, and improve readability

### Style

- [ ] My code follows the established style conventions
- [ ] My code uses native material-ui components/icons/conventions when possible

### Documentation

- [ ] I have commented my code sufficiently, such that an unfamiliar developer could understand my code
- [ ] I have added/updated README's and related documentation, as needed

### Tests

- [ ] I have added/updated unit tests for any code I've added/updated
- [ ] I have added/updated the Postman requests/tests to account for any API endpoints I've added/updated

### Linting/Formatting

- [ ] I have run the linter and fixed any issues, as needed  
       _See the `lint` commands in package.json_
- [ ] I have run the formatter and fixed any issues, as needed  
       _See the `format` commands in package.json_

### SonarCloud

- [ ] I have addressed all SonarCloud Bugs, Vulnerabilities, Security Hotspots, and Code Smells

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Screenshots

Please add any relevant UI screenshots, if applicable.

### CODE REVIEW

**Comment by rstens** [comment]
Sonarqube is griping about the regex - it states that you should use \d instead of [0-9] or [1-2]? instead of [1-2]{0,1}

File: app/src/components/map/OccurrenceFeatureGroup.tsx [hunk 1]
```diff
@@ -42,25 +42,23 @@ const OccurrenceFeatureGroup: React.FC<IOccurrenceFeatureGroupProps> = (props) =
...
-                key={occurrence.occurrenceId}
-                position={[(geometry.geometry as Point).coordinates[1], (geometry.geometry as Point).coordinates[0]]}>
-                <OccurrenceFeaturePopup featureData={featureData} />
-              </Marker>
-            );
-          })}
+          return (
+            <Marker
+              key={occurrence.occurrenceId}
+              position={[(geometry.geometry as Point).coordinates[1], (geometry.geometry as Point).coordinates[0]]}>
```
Comment (anissa-agahchen) [inline app/src/components/map/OccurrenceFeatureGroup.tsx]:
```suggestion
             position={[(geometry?.geometry as Point).coordinates[1], (geometry?.geometry as Point).coordinates[0]]}>
```

Comment (anissa-agahchen) [reply app/src/components/map/OccurrenceFeatureGroup.tsx]: Nit: spelling of this file should be

database/src/migrations/20211018010922_update_funding_source.ts

File: api/src/utils/spatial-utils.ts [hunk 2]
```diff
@@ -12,7 +12,8 @@ const SOPUTH_UTM_BASE_ZONE_NUMBER = 32700;
 const NORTH_UTM_ZONE_LETTERS = ['N', 'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X'];
 const SOUTH_UTM_ZONE_LETTERS = ['C', 'D', 'E', 'F', 'G', 'H', 'J', 'K', 'L', 'M'];
 
-const UTM_STRING_FORMAT = RegExp(/^[1-9]{0,1}[0-9][NPQRSTUVWXCDEFGHJKLM] [0-9]+ [0-9]+$/i);
+const UTM_STRING_FORMAT = RegExp(/^[1-9]\d?[NPQRSTUVWXCDEFGHJKLM]? \d+ \d+$/i);
+const UTM_ZONE_WITH_LETTER_FORMAT = RegExp(/^[1-9]\d?[NPQRSTUVWXCDEFGHJKLM]$/i);
```
Comment (anissa-agahchen) [inline api/src/utils/spatial-utils.ts:16]: nice!

File: app/src/components/map/OccurrenceFeatureGroup.tsx (hunk 1 above)
Comment (NickPhura) [reply app/src/components/map/OccurrenceFeatureGroup.tsx]: Updated

**Comment by NickPhura** [comment]
> 
> 
> Sonarqube is griping about the regex - it states that you should use \d instead of [0-9] or [1-2]? instead of [1-2]{0,1}

I've made some changes, so it should be happy now

**Review by anissa-agahchen — DISMISSED** [review]

**Review by anissa-agahchen — APPROVED** [review]

**Pull request merged**